### ISSUE TITLE
 error when some package uses 'exports' instead of 'module.exports'

### ISSUE THREAD

**Issue by andrewboot** [standard-things/esm#110]
Hi! First of all, thank you for this awesome tool!

TL;DR: If a module exports `something` using not `module.exports = something`, but `exports = something` (which should be valid in node), the tool seems to not recognize that.

Details:

I've stumbled upon this behavior using `app-root-dir` package, so I'll explain using it as an example.
First, I tried to import it as usual:
`import appRootDir from 'app-root-dir'`
I received an error:
```
UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 1): SyntaxError: Module '<pathToProject>/node_modules/app-root-dir/lib/index.js' does not provide an export named 'default'
```
I checked `app-root-dir`'s source code. It's clearly exports two functions, `get` and `set`, like this:
```
exports.get = function() { ... }
exports.set = function() { ... }
```

So I tried to import just get function:
`import { get } from 'app-root-dir'`

It didn't work, the error:
```
UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 1): SyntaxError: Module '<pathToProject>/node_modules/app-root-dir/lib/index.js' does not provide an export named 'get'
```
Finally, I tried to load module using dynamic `import()` and saw that just empty object was loaded. Then I changed `exports` to `module.exports` and now `import { get } from 'app-root-dir'` works fine. By the way, shouldn't default export work too in this case?

**Comment by jdalton** [comment]
Hi @andrewboot!

bar.js
```js
exports = { "a": 1 }
```

foo.js
```js
var bar = require("./bar")
console.log(bar)
```
then
```
node foo.js
```
will log an empty object, `{}`

**Comment by andrewboot** [comment]
Hey @jdalton
But using your analogue, that module didn't reassign `exports` like
```
exports = { a: 1 }
```
it extended exports object with methods, like
```
exports.a = 1
```
In case of that module, I gave short examples right from the source code:
```
exports.get = function() { ... }
exports.set = function() { ... }
```
Does it still severs the association somehow?

**Comment by jdalton** [comment]
I'm only going off your exact words
> but `exports = something` (which should be valid in node), the tool seems to not recognize that.

If you have a problem it'd be helpful if you would create a simple repro repo so I can take a look.

**Comment by andrewboot** [comment]
Oops, sorry, you're right - I confusingly shortened the real issue to a wrong representation.

As I pointed out early, just by changing `exports.get = ... ` to `module.exports.get = ...` in source code of `app-root-dir`, I managed to solve the problem. Fortunately, writing `exports` instead of `module.exports` is a rare practice these days. Still, every user of this tool will have problems with such libraries.

Are issue symptoms clear for you? I'll create repository if not, as you proposed. And if my assumptions are correct and `exports.get = { ... }` should be imported as `import { get } from 'package-name'` without any problems, then I think it's kind of early to close the issue.

**Comment by jdalton** [comment]
> I'll create repository if not, as you proposed.

Please do. It's just easier. It lets me know what version you're using, which options, how you're invoking it, the file names, things like whitespace, various deps and config, etc.

**Comment by andrewboot** [comment]
@jdalton 
Done: https://github.com/andrewboot/std-esm-error
Btw, I use Node 8.5.0 on Linux

**Comment by jdalton** [comment]
Ah ok!

So you're mixing options a bit. The `"esm": "all"` is saying treat all files as ESM, but then you're exposing bits of CJS into ESM with the `"cjs":true` option. In this case your ESM module isn't exporting anything and is relying on the CJS `exports` object. If using `esm` of `"all"` it'd make sense to import and export as ESM does. Regardless there is a bug here, the `exports` object that is exposed when `esm` is `all` and `cjs` is `true` is the wrong one.

**Comment by andrewboot** [comment]
Awesome, thank you!

**Comment by jdalton** [comment]
v0.11.1 is released 🎉
